scipy/cluster/tests/test_hierarchy.py

```diff
@@ -423,45 +456,43 @@ def help_is_isomorphic_randperm(self, nobs, nclusters, noniso=False, nerrors=0,
             assert is_isomorphic(b, a) == (not noniso)
 
 
-@skip_xp_backends(cpu_only=True)
 class TestIsValidLinkage:
 
-    def test_is_valid_linkage_various_size(self, xp):
-        for nrow, ncol, valid in [(2, 5, False), (2, 3, False),
-                                  (1, 4, True), (2, 4, True)]:
-            self.check_is_valid_linkage_various_size(nrow, ncol, valid, xp)
-
-    def check_is_valid_linkage_various_size(self, nrow, ncol, valid, xp):
+    @pytest.mark.parametrize("nrow, ncol, valid", [(2, 5, False), (2, 3, False),
+                                                  (1, 4, True), (2, 4, True)])
+    def test_is_valid_linkage_various_size(self, nrow, ncol, valid, xp):
         # Tests is_valid_linkage(Z) with linkage matrices of various sizes
         Z = xp.asarray([[0, 1, 3.0, 2, 5],
                         [3, 2, 4.0, 3, 3]], dtype=xp.float64)
         Z = Z[:nrow, :ncol]
-        assert_(is_valid_linkage(Z) == valid)
+        xp_assert_equal(is_valid_linkage(Z), valid, check_namespace=False)
         if not valid:
             assert_raises(ValueError, is_valid_linkage, Z, throw=True)
 
     def test_is_valid_linkage_int_type(self, xp):
         # Tests is_valid_linkage(Z) with integer type.
         Z = xp.asarray([[0, 1, 3.0, 2],
                         [3, 2, 4.0, 3]], dtype=xp.int64)
-        assert_(is_valid_linkage(Z) is False)
+        xp_assert_equal(is_valid_linkage(Z), False, check_namespace=False)
```

Comment:
This is an artefact of `lazy_xp_function`: `is_valid_linkage` always returns (python bool) False, but `jax.jit(is_valid_linkage)` returns `jnp.Array(False)`. This is not the case when jax.jit is applied higher in the call stack, as it almost always will in real life.